splunk/src/main/java/com/splunk/Service.java

```diff
@@ -1319,7 +1325,16 @@ public Job search(String query, Map<String, Object> args) {
         if (token != null && !cookieStore.hasSplunkAuthCookie() ) {
             request.getHeader().put("Authorization", token);
         }
-        return super.send(fullpath(path), request);
+        ResponseMessage responseMessage = super.send(fullpath(path), request);
+        if(responseMessage.getStatus() == 401 && (this.autologin && this.username!= null && this.password != null)){
+            // Executing re-login to renew the session token.
+            this.login(this.username, this.password);
+            responseMessage = super.send(fullpath(path), request);

```

Comment:
Per the superclass, on ` status >= 400`, this will throw an exception. In such a case, the next line will likely never run. See class method.
```java
// com.splunk.HttpService
public ResponseMessage send(String path, RequestMessage request)
```

Reply:
in the superclass, we have added a check for status=401 which returns the response back here and so the next lines of code will execute. Ref - https://github.com/splunk/splunk-sdk-java/blob/new-autologin-feature/splunk/src/main/java/com/splunk/HttpService.java#L529
```java
// If user session has expired check for 'autologin' flag to either re-login or throw HTTPException
  if(this instanceof Service && status == 401){
      return new ResponseMessage(401, input);
  }
```

Reply:
I was referring to line 1332, If the login request fails and returns a `status>=400` from the superclass, that would throw an exception if I understand. Otherwise, that's fine, for line 1333. LGTM.